scylla/src/transport/execution_profile.rs

```diff
@@ -191,8 +191,8 @@ pub(crate) mod defaults {
     pub(crate) fn load_balancing_policy() -> Arc<dyn LoadBalancingPolicy> {
         Arc::new(load_balancing::DefaultPolicy::default())
     }
-    pub(crate) fn retry_policy() -> Box<dyn RetryPolicy> {
-        Box::new(DefaultRetryPolicy::new())
+    pub(crate) fn retry_policy() -> Arc<dyn RetryPolicy> {
+        Arc::new(DefaultRetryPolicy::new())
```

Comment:
We do lose _something_. Namely, `Box` has an optimisation which makes it not allocate anything for ZSTs, and `Arc` allocates always (its control block). However, I don't know if the optimisation is viable for `Box<dyn Trait>` with ZST inside.